src/test/java/bio/terra/service/common/gcs/GcsUriUtilsTest.java

```diff
@@ -0,0 +1,80 @@
+package bio.terra.service.common.gcs;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThrows;
+
+import bio.terra.common.category.Unit;
+import com.google.cloud.storage.BlobId;
+import org.apache.commons.lang3.StringUtils;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+@Category(Unit.class)
+public class GcsUriUtilsTest {
+
+  @Test
+  public void testGsPathExtraction() {
+    assertThat(
+        GcsUriUtils.parseBlobUri("gs://mybucket/my/path.txt"),
+        equalTo(BlobId.of("mybucket", "my/path.txt")));
+  }
+
+  @Test
+  public void testGsPathExtractionSpecialCharacters() {
+    assertThat(
+        GcsUriUtils.parseBlobUri("gs://mybucket/my/path# space %.txt"),
+        equalTo(BlobId.of("mybucket", "my/path# space %.txt")));
+  }
+
+  @Test
+  public void testGsPathExtractionBucketTooShort() {
+    Exception exception =
+        assertThrows(IllegalArgumentException.class, () -> GcsUriUtils.parseBlobUri("gs://a/b"));
+    assertEquals("Invalid bucket name in gs path: 'gs://a/b'", exception.getMessage());
+  }
+
+  @Test
+  public void testGsPathExtractionBucketTooLong() {
+    final String bucketName = StringUtils.rightPad("abc", 250, 'x');
```

Comment:
This use of `StringUtils` is unnecessary, `String.repeat()` can be used here.
```suggestion
    final String bucketName = "x".repeat(250);
```